### PR TITLE
Feature/sg 251 refactor cifar10 dataset interface

### PR DESCRIPTION
**Pull request by spsancti**

### CODE REVIEW

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml [hunk 1]
```diff
@@ -1,2 +1,54 @@
-batch_size: 256 # batch size for trainset
-val_batch_size: 512 # batch size for valset in DatasetInterface
\ No newline at end of file
+-batch_size: 256 # batch size for trainset
```
Comment (shaydeci) [inline src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:1]: Why the "-" ?

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml [hunk 2]
```diff
@@ -1,2 +1,54 @@
...
+-val_batch_size: 512 # batch size for valset in DatasetInterface
```
Comment (shaydeci) [inline src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:2]: Why the "-" ?

File: src/super_gradients/training/datasets/classification_datasets/cifar.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+from torchvision.transforms import Compose
+
+from common.factories.list_factory import ListFactory
+from common.factories.transforms_factory import TransformsFactory
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from torchvision.datasets import CIFAR10, CIFAR100
+
+
+class Cifar10(CIFAR10):
```
Comment (shaydeci) [inline src/super_gradients/training/datasets/classification_datasets/cifar.py:11]: Lets add some docs, we need to at least clarify the transforms argument since its not in the parent class.

File: src/super_gradients/training/datasets/classification_datasets/cifar.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            root=root,
+            train=train,
+            transform=Compose(transforms),
+            target_transform=target_transform,
+            download=download,
+        )
+
+
+class Cifar100(CIFAR100):
+    @resolve_param("transforms", ListFactory(TransformsFactory()))
```
Comment (shaydeci) [inline src/super_gradients/training/datasets/classification_datasets/cifar.py:31]: Same comment about docs as above.

File: tests/end_to_end_tests/cifar_trainer_test.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+from super_gradients.training.datasets.dataset_interfaces import LibraryDatasetInterface
+from super_gradients.training.dataloaders.dataloader_factory import (
+    cifar10_train,
+    cifar10_val,
+    cifar100_train,
+    cifar100_val,
+)
+import torch
+
+torch.multiprocessing.set_sharing_strategy("file_system")
```
Comment (shaydeci) [inline tests/end_to_end_tests/cifar_trainer_test.py:17]: Whats this for ?

**Review by shaydeci — CHANGES_REQUESTED** [review]
Nice!
Just some minor comments/clarifications.

File: src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
+-batch_size: 256 # batch size for trainset
```
[the remark this answers is not in the document]
Comment (spsancti) [reply src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml:1]: Was there originally, will remove if it doesn't break tests

File: src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+-val_batch_size: 512 # batch size for valset in DatasetInterface
```
[the remark this answers is not in the document]
Comment (spsancti) [reply src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml:2]: Was there originally, will remove if it doesn't break tests

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml (hunk 1 above)
Comment (spsancti) [reply src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:1]: Was there originally, will remove if it doesn't break tests

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml (hunk 2 above)
Comment (spsancti) [reply src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:2]: Was there originally, will remove if it doesn't break tests

File: tests/end_to_end_tests/cifar_trainer_test.py (hunk 5 above)
Comment (spsancti) [reply tests/end_to_end_tests/cifar_trainer_test.py:17]: Forgot from debug, removing

File: src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  train: True
+  transforms:
+    - RandomCrop:
+        size: 24
+        padding: 4
+    - RandomHorizontalFlip
+    - ToTensor
+    - Normalize:
+        mean:
+          - 0.4914
```
Comment (shaydeci) [inline src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml:17]: These are the default transforms for cifar10 not cifar100 (take a look at the corresponding dataset interface class and their default parameters defined in LIBRARY_DATASETS).

File: src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+val_batch_size: 512 # batch size for valset in DatasetInterface
+
+# TODO: REMOVE ABOVE, HERE FOR COMPATIBILITY UNTIL WE REMOVE DATASET_INTERFACE
+
+train_dataset_params:
+  root: /data/cifar100
+  train: True
+  transforms:
+    - RandomCrop:
+        size: 24
```
Comment (shaydeci) [inline src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml:11]: Default crop size is 32 not 24.

File: src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  pin_memory: True
+
+val_dataset_params:
+  root: /data/cifar100
+  train: False
+  transforms:
+    - ToTensor
+    - Normalize:
+        mean:
+          - 0.4914
```
Comment (shaydeci) [inline src/super_gradients/recipes/dataset_params/cifar100_dataset_params.yaml:40]: Again, wrong normalisation values (cifar10 not 100)

File: src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml [hunk 11]
```diff
@@ -1,2 +1,54 @@
...
+val_batch_size: 512 # batch size for valset in DatasetInterface
+
+# TODO: REMOVE ABOVE, HERE FOR COMPATIBILITY UNTIL WE REMOVE DATASET_INTERFACE
+
+train_dataset_params:
+  root: /data/cifar10
+  train: True
+  transforms:
+    - RandomCrop:
+        size: 24
```
Comment (shaydeci) [inline src/super_gradients/recipes/dataset_params/cifar10_dataset_params.yaml:11]: Wrong default crop size (should be 32).

**Review by shaydeci — CHANGES_REQUESTED** [review]
Nice work. Just got some some values wrong in the transforms inside the yaml files. Please fix them and we can merge.

**Review by shaydeci — APPROVED** [review]
LGTM

**Review by shaydeci — APPROVED** [review]
LGTM

**Review by shaydeci — APPROVED** [review]

**Review by shaydeci — APPROVED** [review]
